File: javascript/leetcode/implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (needle == '') { return 0 }
    var i = 0
    while (i < haystack.length + 1 - needle.length) {
        if (haystack[i] == needle[0]) {
            j = 0
            originalI = i
            while (j < needle.length) {
                if (needle[j] == haystack[i] ) {
                    j++
                    i++
                } else {
                    break
                }
            }
            if (j == needle.length) {
                return originalI
            } else {
                i = originalI
            }
        }
        i++
    }
    return -1
};

File: python/leetcode/implement-strstr.py
class Solution:
    def strStr(self, haystack, needle):
        if needle == '':
            return 0
        i = 0
        while i < len(haystack) - (len(needle) - 1):
            if haystack[i] == needle[0]:
                j = 0
                original_i = i
                while (j < len(needle)):
                    if (needle[j] == haystack[i]):
                        j += 1
                        i += 1
                    else: 
                        break
                if j == len(needle):
                    return original_i
                else:
                    i = original_i
            i += 1
        return -1